homeassistant/components/opower/coordinator.py

```diff
@@ -50,7 +50,7 @@ def __init__(
             update_interval=timedelta(hours=12),
         )
         self.api = Opower(
-            aiohttp_client.async_get_clientsession(hass),
+            aiohttp_client.async_create_clientsession(hass),
```

Comment:
So, why is that?

Reply:
Not quite sure. The 2 core bugs fail for the same reason. The library fails to parse the login page but only while getting updated data, not in the config flow. The only difference is the config flow creates its own session while coordinator uses a common session. I suspect it's because the library modifies the default user agent instead of specifying it on every HTTP request.

Reply:
> I suspect it's because the library modifies the default user agent instead of specifying it on every HTTP request.

That sounds familiar! Got a pointer to me for that upstream?

Reply:
https://github.com/tronikos/opower/blob/main/src/opower/opower.py#L136

Reply:
Aah yes, that shouldn't be used/set and, indeed passed on request.
We've "monkeypatched" it in Home Assistant to prevent it to work, as modifying internals of shared sessions like that causes issues (and has caused many issues in the past).

Ref: <https://github.com/home-assistant/core/blob/a8a3b6778ef2d2cf146319d28a55d13e80a9a182/homeassistant/helpers/aiohttp_client.py#L148>

TBH, from that perspective, I don't agree on the solution provided in this PR.

../Frenck

Reply:
I modified the library to pass the headers in each request and created https://github.com/home-assistant/core/pull/97548 to bump the dependency